presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/execution/SimpleAddressSqlFunctionLanguageConfigSpec.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.functionNamespace.execution;
+
+import com.facebook.presto.spi.function.FunctionImplementationType;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.facebook.presto.spi.function.FunctionImplementationType.SQL;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class SimpleAddressSqlFunctionLanguageConfigSpec
+{
+    private final FunctionImplementationType implementationType;
+    private final Optional<String> addresses;
+    private final boolean retrySameAddress;
+
+    @JsonCreator
+    public SimpleAddressSqlFunctionLanguageConfigSpec(
+            @JsonProperty("implementation-type") FunctionImplementationType implementationType,
+            @JsonProperty("addresses") Optional<String> addresses,
+            @JsonProperty("retry-same-address") boolean retrySameAddress)
```

Comment:
We now have a nested config represented by JSON, and those config are actually defined to be passed in as part of drift `SimpleAddressSelectorConfig`. I think we can improve how the configuration are specified here.

For each language there will be a single required config `implementationType`. e.g.
```
sql.implementation-type=build-in
java.implementation-type=thrift
```
Then depending on the value of the implementation type, we bind additional config in the module. e.g.
```
java.thrift.address=....
java.thrift.retry-same-address=...
```

This would make the config more readable.